diff --git a/build/common.sh b/build/common.sh
index 6402b1219ea..90b9e965ad2 100755
--- a/build/common.sh
+++ b/build/common.sh
@@ -78,7 +78,7 @@ readonly REMOTE_OUTPUT_GOPATH="${REMOTE_OUTPUT_SUBPATH}/go"
 
 # These are the default versions (image tags) for their respective base images.
 readonly __default_distroless_iptables_version=v0.9.1
-readonly __default_go_runner_version=v2.4.0-go1.26.2-bookworm.0
+readonly __default_go_runner_version=v0.21.1
 readonly __default_setcap_version=bookworm-v1.0.6
 
 # The default image for all binaries which are dynamically linked.
diff --git a/build/pause/Makefile b/build/pause/Makefile
index c2446d8cd6b..b022442e37e 100644
--- a/build/pause/Makefile
+++ b/build/pause/Makefile
@@ -20,7 +20,7 @@ IMAGE = $(REGISTRY)/pause
 TAG ?= 3.10.2
 REV = $(shell git describe --contains --always --match='v*')
 
-# Architectures supported: amd64, arm, arm64, ppc64le and s390x
+# Architectures supported: amd64, arm, arm64, ppc64le, s390x and loong64
 ARCH ?= amd64
 # Operating systems supported: linux, windows
 OS ?= linux
@@ -38,7 +38,7 @@ BASE := ${BASE.${OS}}
 JQ_IMAGE := ghcr.io/jqlang/jq@sha256:a186dcd84a1e28bb48cdf3d7768b890d08621a87bb651fadb7db6815a6bf5ad5
 
 ALL_OS = linux windows
-ALL_ARCH.linux = amd64 arm arm64 ppc64le s390x
+ALL_ARCH.linux = amd64 arm arm64 ppc64le s390x loong64
 ALL_OS_ARCH.linux = $(foreach arch, ${ALL_ARCH.linux}, linux-$(arch))
 ALL_ARCH.windows = amd64
 # ALL_OSVERSIONS lists all os.versions in BASE.windows.
@@ -73,6 +73,7 @@ TRIPLE.linux-arm := arm-linux-gnueabihf
 TRIPLE.linux-arm64 := aarch64-linux-gnu
 TRIPLE.linux-ppc64le := powerpc64le-linux-gnu
 TRIPLE.linux-s390x := s390x-linux-gnu
+TRIPLE.linux-loong64 := loongarch64-linux-gnu
 TRIPLE := ${TRIPLE.${OS}-${ARCH}}
 
 # If you want to build AND push all containers, see the 'all-push' rule.
@@ -104,6 +105,12 @@ sub-container-%:
 
 build: $(foreach binary, ${BIN}, bin/${binary}-${OS}-${ARCH})
 
+ifneq ($(USE_LOCAL_CC),)
+bin/${BIN.linux}-$(OS)-$(ARCH): $(SRCS)
+	mkdir -p bin
+	$(TRIPLE)-gcc $(CFLAGS) -o $@ $^
+	$(TRIPLE)-strip $@
+else
 bin/${BIN.linux}-$(OS)-$(ARCH): $(SRCS)
 	mkdir -p bin
 	docker run --rm -u $$(id -u):$$(id -g) -v $$(pwd):/build \
@@ -112,6 +119,7 @@ bin/${BIN.linux}-$(OS)-$(ARCH): $(SRCS)
 			cd /build && \
 			$(TRIPLE)-gcc $(CFLAGS) -o $@ $^ && \
 			$(TRIPLE)-strip $(foreach binary, $@, ${binary}${EXTENSION})"
+endif
 
 bin/wincat-windows-${ARCH}: windows/wincat/wincat.go
 	mkdir -p bin
diff --git a/hack/lib/golang.sh b/hack/lib/golang.sh
index 95c775f8f70..9fd49c2f9f6 100755
--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -25,6 +25,7 @@ readonly KUBE_SUPPORTED_SERVER_PLATFORMS=(
   linux/arm64
   linux/s390x
   linux/ppc64le
+  linux/loong64
 )
 
 # The node platforms we build for
@@ -33,6 +34,7 @@ readonly KUBE_SUPPORTED_NODE_PLATFORMS=(
   linux/arm64
   linux/s390x
   linux/ppc64le
+  linux/loong64
   windows/amd64
 )
 
@@ -45,6 +47,7 @@ readonly KUBE_SUPPORTED_CLIENT_PLATFORMS=(
   linux/arm64
   linux/s390x
   linux/ppc64le
+  linux/loong64
   darwin/amd64
   darwin/arm64
   windows/amd64
@@ -59,6 +62,7 @@ readonly KUBE_SUPPORTED_TEST_PLATFORMS=(
   linux/arm64
   linux/s390x
   linux/ppc64le
+  linux/loong64
   darwin/amd64
   darwin/arm64
   windows/amd64
@@ -212,8 +216,8 @@ kube::golang::setup_platforms() {
 
   elif [[ "${KUBE_FASTBUILD:-}" == "true" ]]; then
     host_arch=$(kube::util::host_arch)
-    if [[ "${host_arch}" != "amd64" && "${host_arch}" != "arm64" && "${host_arch}" != "ppc64le" && "${host_arch}" != "s390x" ]]; then
-      # on any platform other than amd64, arm64, ppc64le and s390x, we just default to amd64
+    if [[ "${host_arch}" != "amd64" && "${host_arch}" != "arm64" && "${host_arch}" != "ppc64le" && "${host_arch}" != "s390x" && "${host_arch}" != "loong64" ]]; then
+      # on any platform other than amd64, arm64, ppc64le, s390x and loong64, we just default to amd64
       host_arch="amd64"
     fi
     KUBE_SERVER_PLATFORMS=("linux/${host_arch}")
@@ -507,6 +511,10 @@ kube::golang::set_platform_envs() {
         export CGO_ENABLED=1
         export CC=${KUBE_LINUX_S390X_CC:-s390x-linux-gnu-gcc}
         ;;
+      "linux/loong64")
+        export CGO_ENABLED=1
+        export CC=${KUBE_LINUX_LOONG64_CC:-loongarch64-linux-gnu-gcc}
+        ;;
     esac
   fi
 
diff --git a/hack/lib/util.sh b/hack/lib/util.sh
index ed067cc7a30..f08121e811a 100755
--- a/hack/lib/util.sh
+++ b/hack/lib/util.sh
@@ -185,8 +185,11 @@ kube::util::host_arch() {
     ppc64le*)
       host_arch=ppc64le
       ;;
+    loongarch64*)
+      host_arch=loong64
+      ;;
     *)
-      kube::log::error "Unsupported host arch. Must be x86_64, 386, arm, arm64, s390x or ppc64le."
+      kube::log::error "Unsupported host arch. Must be x86_64, 386, arm, arm64, s390x, ppc64le or loongarch64."
       exit 1
       ;;
   esac
diff --git a/test/conformance/image/Makefile b/test/conformance/image/Makefile
index 61a963f6352..3d02347688e 100644
--- a/test/conformance/image/Makefile
+++ b/test/conformance/image/Makefile
@@ -34,7 +34,7 @@ CLUSTER_DIR?=$(shell pwd)/../../../cluster/
 # This is defined in root Makefile, but some build contexts do not refer to them
 KUBE_BASE_IMAGE_REGISTRY?=registry.k8s.io
 BASE_IMAGE_VERSION?=bookworm-v1.0.6
-RUNNERIMAGE?=${KUBE_BASE_IMAGE_REGISTRY}/build-image/debian-base-${ARCH}:${BASE_IMAGE_VERSION}
+RUNNERIMAGE?=${KUBE_BASE_IMAGE_REGISTRY}/debian-base-${ARCH}:${BASE_IMAGE_VERSION}
 
 TEMP_DIR:=$(shell mktemp -d -t conformance-XXXXXX)
 
